### DISCUSSION TITLE
Finger print the favicon?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#3416] [Q&A] [answered]
I generally finger print all my assets with a hash, ideally I'd like to to do this for the favicon. But, I cannot find where Remix is rendering the favicon tag. I assume it's somewhere in the `Meta` component?

**Comment by machour** [comment]
Remix isn't rendering a favicon, so the browser is defaulting to /favicon.ico (and it finds the one provided in public/)

You can override this by providing your own favicon using the `meta()` function, but I'm not sure this will automatically fingerprint it for you.

**Comment by kiliman** [comment, marked as answer]
```ts
import favicon from '~/assets/favicon.png'
export const links: LinksFunction = () => [
  { rel: 'icon', type: 'image/png', href: favicon }
]
```
https://www.w3.org/2005/10/howto-favicon
